src/services/api/api.controller.ts

```diff
@@ -301,31 +311,19 @@ export class ApiController {
     type: VotePostResponseDto,
   })
   @UseGuards(AuthGuard('jwt'))
-  @UseGuards(RequireHiveVerify)
+  @UseInterceptors(UserDetailsInterceptor)
   @Post(`/hive/vote`)
-  async votePost(@Body() data: VotePostDto) {
-    const { author, permlink } = data;
-    // const delegatedAuth = await this.delegatedAuthorityRepository.findOne({
-    //   to: 'threespeak.beta',
-    //   from:
-    // })
-    // TODO: get hive username from auth
-    const delegatedAuth = true;
-    const voter = 'vaultec';
-    if (delegatedAuth) {
-      try {
-        // console.log(out)
-        return this.hiveChainRepository.vote({ author, permlink, voter, weight: 500 });
-      } catch (ex) {
-        console.log(ex);
-        console.log(ex.message);
-        throw new BadRequestException(ex.message);
-      }
-      // await appContainer.self
-    } else {
-      throw new BadRequestException(`Missing posting autority on HIVE account 'vaultec'`, {
-        description: 'HIVE_MISSING_POSTING_AUTHORITY',
-      });
-    }
+  async votePost(@Body() data: VotePostDto, @Request() req: any) {
+    const parsedRequest = parseAndValidateRequest(req, this.#logger);
+    const { author, permlink, weight } = data;
+
+    return await this.hiveService.vote({
+      sub: parsedRequest.user.sub,
+      votingAccount: parsedRequest.user.username,
```

Comment:
Hmm... What happens if a DID user is logged in?

Reply:
good spot